coredb-operator/src/statefulset.rs

```diff
@@ -228,15 +228,78 @@ pub async fn reconcile_sts(cdb: &CoreDB, ctx: Arc<Context>) -> Result<(), Error>
     let sts: StatefulSet = stateful_set_from_cdb(cdb);
 
     let sts_api: Api<StatefulSet> = Api::namespaced(client, &sts.clone().metadata.namespace.unwrap());
+    // If cdb is running and storage is resized in CoreDB Custom Resource then follow steps to resize PVC
+    // Reference Article: https://itnext.io/resizing-statefulset-persistent-volumes-with-zero-downtime-916ebc65b1d4
+    if cdb.status.is_some()
+        && cdb.status.clone().unwrap().running
+        && cdb.status.clone().unwrap().storage != cdb.spec.storage
+    {
+        delete_sts(ctx.client.clone(), &sts).await;
+        update_pvc(ctx.client.clone(), &sts, &cdb).await;
+    }
 
     let ps = PatchParams::apply("cntrlr").force();
     let _o = sts_api
         .patch(&sts.clone().metadata.name.unwrap(), &ps, &Patch::Apply(&sts))
         .await
         .map_err(Error::KubeError)?;
+
     Ok(())
 }
 
+async fn delete_sts(client: Client, sts: &StatefulSet) {
+    let sts_api: Api<StatefulSet> = Api::namespaced(client, &sts.clone().metadata.namespace.unwrap());
+
+    let sts_name = sts.clone().metadata.name.unwrap();
+
+    let delete_params: DeleteParams = DeleteParams {
+        dry_run: false,
+        grace_period_seconds: None,
+        propagation_policy: Some(kube::api::PropagationPolicy::Orphan),
+        preconditions: None,
+    };
+
+    let _o = sts_api
+        .delete(&sts_name, &delete_params)
+        .await
+        .map_err(Error::KubeError);
+}
+
+async fn update_pvc(client: Client, sts: &StatefulSet, cdb: &CoreDB) {
+    let pvc_api: Api<PersistentVolumeClaim> =
+        Api::namespaced(client, &sts.clone().metadata.namespace.unwrap());
+
+    let sts_name = sts.clone().metadata.name.unwrap();
+    let pvc_name = format!("data-{}-0", sts_name);
```

Comment:
Maybe there is a better way to query the PVC (i.e. without the "-0"). Need to take a look at it